### DISCUSSION TITLE
Basic Questions regarding UI and business logic separation

### DISCUSSION THREAD

**Discussion by haaris94** [jonataslaw/getx#2166] [Q&A]
I have the following questions

1. Is it ok to initialize `Controllers` inside the `build` method of the `Stateless widget`, is it bad for performance?
2. If I initialize the controllers outside the build method, it gives me an error 'controller is initialized with non-nullable variable'.
3. If I want to update the value of the `TextField` using `TextEditingController` from the `Controller` class, most tutorial advice to create the `TextEditingController` in the `Controller` class, isn't it bad to bring UI code to the `Controller` class. is it good practice?
4. Is there any way to trigger a method like `showSnackbar()` in View class(Stateless Widget) from the `Controller` when the value of the `msg.obs` changes in the `Controller` class

Thank you. I am new to this package.

**Comment by ashunasar** [comment]
can you please share the code
